fiducial_slam/src/map.cpp

```diff
@@ -96,21 +96,79 @@ static void updateTransform(tf2::Transform &t1, double var1,
 }
 #endif
 
+/* 
+*  Takes 2 variances and gives the kalman gain figure
+*  Kalmain gain represents how much to trust the new measurement compared to the existing one
+*  Taken from equation 12 https://www.bzarg.com/p/how-a-kalman-filter-works-in-pictures/ 
+*/  
+static double kalman_gain(double var1, double var2) {
+    return var1 / (var1 + var2);
+}
+
+
+static double probabiltyAtPoint(const double x, const double u, const double var) {
+    return (1.0 /(sqrt(var) * sqrt(2.0 *M_PI)))* exp (-( (x-u)*(x-u) )/(2.0 * var));
+}
+
+/*
+ * Takes in 2 estimates and the combined estimate and calculates the
+ * variance of the new estimate to normalize it
+ */
+static double normalizeDavid(const double newMean,
+		             const double mean1, double var1,
+			     const double mean2, double var2) {
+    
+    // Find the probabilities of the new mean in both original gaussians
+    double prob1_at_newMean = probabiltyAtPoint(newMean, mean1, var1);
+    double prob2_at_newMean = probabiltyAtPoint(newMean, mean2, var2);
+    // We use the sum in quadrature of these 2 probabilities 
+    double prob_at_newMean = sqrt(pow(prob1_at_newMean, 2) + pow(prob2_at_newMean, 2));
```

Comment:
is it more efficient to do `x * x` than `std::pow(x, 2)`?

Reply:
With any amount of optimization (-O1) they compile to the same thing.
https://gcc.godbolt.org/z/_QIozo

Reply:
That's good to know.  We are not using `std::pow` and `std::sqrt` here though.